### ISSUE TITLE
Daterangepicker not correctly localized

### ISSUE THREAD

**Issue by psychowood** [firefly-iii/firefly-iii#3680]
Hi,
Daterangepicker seems to be not localized correctly, at least for me: [it](https://github.com/firefly-iii/firefly-iii/blob/2c69a2eda2bfda39bde1eafd2d314a26f8d0d928/public/v1/js/ff/firefly.js#L43) is [hardcoded](https://github.com/firefly-iii/firefly-iii/blob/2c69a2eda2bfda39bde1eafd2d314a26f8d0d928/public/v1/js/ff/reports/index.js#L40) to `'YYYY-MM-DD'`

Since I'm arealdy here... Datepicker always shows mm/dd/yyyy, regardless of language and locale (it_IT in my case); I tried forcing DEFAULT_LOCALE too, without success. It could be related to #1133 since I'm using Firefox on macOS.
Would you consider using a custom datepicker, as a reeeeeally minor enhancement to the UI? AdminLTE should be already bundled with [tempusdominus](https://github.com/ColorlibHQ/AdminLTE/tree/e78ee8d9caf59e55880488dfbf10901ecd4bfbd7/plugins/tempusdominus-bootstrap-4) that could help in that regard.

---
Debug information generated at 2020-08-09 23:02:47 Europe/Rome for Firefly III version **5.3.3**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.3.3 |
| Firefly III API | 1.3.0 |
| PHP | 7.4.8 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | 3c9524c2-6757-4aec-8d85-5387234150a5 |
| Using docker? | true |
| Telemetry | false |
| Layout | v1 |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | it_IT |
| Default locale | it_IT |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Login provider | eloquent  |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| Session start | 2020-08-01 00:00:00 |
| Session end | 2020-08-31 23:59:59 |
| Session first | 2008-02-08 00:00:00 |
| User ID | 1 |
| User language | it_IT |
| User locale | it_IT |
| Attempt at "it_IT.utf8" | 'it_IT.utf8' |
| Attempt at "it_IT.UTF-8" | 'it_IT.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:78.0) Gecko/20100101 Firefox/78.0 |

**Comment by JC5** [comment]
That value is for internal use only. It's intentional.

**Comment by JC5** [comment]
> Would you consider using a custom datepicker, as a reeeeeally minor enhancement to the UI?  

Sorry, I missed this. No.

**Comment by psychowood** [comment]
> That value is for internal use only. It's intentional.

I opened the issue because I noticed that format in the Report page:

<img width="855" alt="image" src="https://user-images.githubusercontent.com/115389/89759975-95959780-daeb-11ea-9e77-68c37b4c2c96.png">

is this another browser related issue?

Regarding the datepicker, everything is already set to it_IT, I even added some extensions... it probably inherits directly the OS language, that's the only part in english. Who care, I'll live with it.

Thanks!
